acai-ios/ViewControllers/LoginViewController.swift

```diff
@@ -122,71 +108,184 @@ class LoginViewController: UIViewController {
+        NotificationCenter.default.addObserver(self, selector: #selector(hideKeyboard), name: UIResponder.keyboardWillHideNotification, object: nil)
+    }
+
+    func resetTextFields() {
+        email = ""
+        firstName = ""
+        lastName = ""
+        password = ""
+        phoneNumber = ""
+    }
+
+    @objc func showKeyboard(notification: NSNotification) {
+        let keyboardFrame = notification.userInfo?[UIResponder.keyboardFrameEndUserInfoKey] as! CGRect
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
+            self.tableView.snp.remakeConstraints { make in
+                make.leading.top.trailing.equalTo(self.view.safeAreaLayoutGuide)
+                make.bottom.equalToSuperview().offset(-keyboardFrame.height)
             }
         }
     }
 
-    @objc func signUp() {
-        if signUpButton.type == .switchDataEntry {
-            loginButton.toggleColor()
-            signUpButton.toggleColor()
-            switchDataEntry(to: .signUp)
-        } else {
-            guard let email = emailInputView.textField.text, let name = nameInputView.textField.text, let password = passwordInputView.textField.text else { return }
-            emailInputView.invalidEntryLabel.isHidden = email.isValidEmail()
-            nameInputView.invalidEntryLabel.isHidden = name.isValidName()
-            passwordInputView.invalidEntryLabel.isHidden = password.isValidPassword()
-            if email.isValidEmail() && name.isValidName() && password.isValidPassword() {
-                createUser(name: name, email: email, password: password)
+    @objc func hideKeyboard(notification: NSNotification) {
+        tableView.snp.remakeConstraints { make in
+            make.leading.trailing.top.equalTo(view.safeAreaLayoutGuide)
+            make.bottom.equalToSuperview()
+        }
+    }
+
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        NotificationCenter.default.removeObserver(self)
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+    }
+
+}
+
+extension LoginViewController: SelectRoundedButtonDelegate {
+    func register() {
+        attemptedNetworking = true
+        (tableView.cellForRow(at: IndexPath(row: 0, section: 0)) as? InputTableViewCell)?.invalidEntryLabel.isHidden = firstName.isValidName()
+        (tableView.cellForRow(at: IndexPath(row: 1, section: 0)) as? InputTableViewCell)?.invalidEntryLabel.isHidden = lastName.isValidName()
+        (tableView.cellForRow(at: IndexPath(row: 2, section: 0)) as? InputTableViewCell)?.invalidEntryLabel.isHidden = phoneNumber.isValidPhoneNumber()
+        (tableView.cellForRow(at: IndexPath(row: 3, section: 0)) as? InputTableViewCell)?.invalidEntryLabel.isHidden = email.isValidEmail()
+        (tableView.cellForRow(at: IndexPath(row: 4, section: 0)) as? InputTableViewCell)?.invalidEntryLabel.isHidden = password.isValidPassword()
```

Comment:
lines 212-215 is sort of hard to read. can we have unwrap each one
```
if let cell = tableView.cellForRow(at: IndexPath(row: 0, section: 0)) as? InputTableViewCell {
    ...
}
...
```
Also i don't think this handles the case where the cell isn't on the screen. You might want to double check that this doesn't cause any issues

Reply:
The array of input cells is hard coded and register() can only be called from the register screen where there are 5 input cells